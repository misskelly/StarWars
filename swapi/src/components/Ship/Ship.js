import React from 'react'

const Ship = () => {
  return (
    <section className='.ship-container'>
    </section>
  )
}

export default Ship;