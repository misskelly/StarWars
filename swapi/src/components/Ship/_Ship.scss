//constants
$pi: 3.14159265359;
// Size (rem) and number (int)
$circumference: 50;
$segments: 20;

@function getArcLength() {
  $degrees: 360 / $segments;
  //  Get radians to calculate arc length
  $radians: $degrees * ($pi / 180);
  //  Arc length = Radius * Radians
  $arc: ($circumference / 2) * $radians;
  @return $arc / 2
}

.ship-container {
  grid-area: ship;
  position: relative;
  // overflow: hidden;
}

#ship {
  width: $circumference + rem;
  height: $circumference + rem;
  margin: 1rem auto;
  position: absolute;
  z-index: 10;
  background: hsl(184, 6%, 47%);
  border-radius: 50%;
  border: 5px solid hsl(184, 6%, 47%);
  // top: 10%;
  left: 50%;
  transform: translateX(-50%); //!!
}

#inner {
  width: 100%;
  height: 100%;
  position: relative;
  background-color: lightgray;
  border-radius: 100%;
  animation: spin infinite 40s linear;
}

.segment {
  width: 0;
  height: 0;
  border-style: solid;
  border-width: ($circumference / 2)+rem getArcLength()+rem 0;
  border-color: transparent;
  position: absolute;
  left: 50%;
  margin-left: -(getArcLength())+rem;
  transform-origin: getArcLength()+rem ($circumference / 2)+rem;

  span {
    display: block;
    margin-top: -(($circumference/2)-3)+rem;
    color: hsla(0, 0%, 0%, 0.8);
    position: relative;
    z-index: 10;
    text-align: center;
    font-size: 10px;
    width: getArcLength()+rem;
    transform: translateX(-50%);

    &:before {
      position: absolute;
      left: 50%;
      top: 2rem;
      transform: translateX(-50%);
      font-size: 1rem;
    }
  }
}

@for $i from 1 through $segments {

  .segment:nth-child(#{$i}) {
    $deg: ((360/$segments)*$i)-(360/$segments);
    transform: rotate($deg+deg);

    span {
      &:before {
        content: "========";
        color: hsl(0, 0%, 66%);
      }
    }
  }

}


.pulse {
  -webkit-animation: blink 1200ms infinite;
  animation: blink 1200ms infinite;
}


@keyframes blink {
  50% {
    border: 9px solid hsl(184, 6%, 47%);
  }
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.dash {
  border-right: 5px dashed rgb(136, 127, 127);
}

