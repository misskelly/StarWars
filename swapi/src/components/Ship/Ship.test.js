import React from 'react';
import Ship from './Ship';
import { shallow } from 'enzyme';



describe('Ship', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( < Ship / > );
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

});