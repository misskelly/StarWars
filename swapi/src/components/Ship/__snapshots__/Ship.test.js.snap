// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ship matches the snapshot 1`] = `
<section
  className="ship-container"
>
  <div
    className="pulse spinWheel"
    id="ship"
  >
    <div
      id="inner"
    >
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
      <div
        className="segment dash"
      >
        <span>
           0 0 0 
        </span>
      </div>
    </div>
  </div>
</section>
`;
